File: 12 Interactive Problem/Code Jam 2022 Round 1A/Equal_Sum.cpp
#include<bits/stdc++.h>
using namespace std;


void sum_of_subset(int weight, int total, int sum, int arr[], int n)
{


}


int main()
{

    int T;
    cin>>T;


    while(T--){
        int N, i, j;
        cin>>N;

        int arr1[N*2+1]={0};

        long long int total=0;

        for(i=1;i<N;i++){
            arr1[i-1] = i;
            total += i;
            cout<<i<<" ";
        }
        arr1[i-1] = i;
        total += i;
        cout<<i<<endl;

        for(j=i;j<2*N;j++){
            cin>>arr1[j];
            total += arr1[j];
        }


        sum_of_subset(0, total, total/2, arr1, N);


    }

    return 0;
}

File: 12 Interactive Problem/Code Jam 2022 Round 1A/Equal_sum.py
from itertools import combinations

def sum_of_subset(li, size, target_sum):
    ck = 0
    for i in range(size+1):
        if ck==1:
            break

        for sub_set in combinations(li, i):

            if sum(sub_set) == target_sum:
                ck = 1
                for j in sub_set:
                    print(j, end=" ")
                print()
                break

def main():
    T = int(input())
    if T== -1:
        return

    while T!=0:
        N = int(input())

        if N== -1:
            return

        total = 0;
        li = []

        for i in range(1,N+1):
            total += i;
            li.append(i)
            print(i, end=" ")
        print()

        temp = list(map(int, input().split()))
        if -1 in temp:
            return
            
        total += sum(temp)

        li += temp


        sum_of_subset(li, N, total/2)



if __name__ == '__main__':
    main()
